### DISCUSSION TITLE
Mypy not detecting invalid pydantic.Field default value

### DISCUSSION THREAD

**Discussion by SupImDos** [pydantic/pydantic#3292] [Question] [answered]
Hi,

I have a question about the following code snippets, regarding statically checking field default values:

`example.py`
```python3
import pydantic

class Model(pydantic.BaseModel):
    example_1: int = "chicken"
    example_2: int = pydantic.Field("turkey")

```

`mypy.ini (default example)`
```ini
[mypy]
plugins = pydantic.mypy

follow_imports = silent
warn_redundant_casts = True
warn_unused_ignores = True
disallow_any_generics = True
check_untyped_defs = True
no_implicit_reexport = True

# for strict mypy: (this is the tricky one :-))
disallow_untyped_defs = True

[pydantic-mypy]
init_forbid_extra = True
init_typed = True
warn_required_dynamic_aliases = True
warn_untyped_fields = True
```

If I run mypy against this file:
```console
$ mypy example.py

example.py:5: error: Incompatible types in assignment (expression has type "str", variable has type "int")
Found 1 error in 1 file (checked 1 source file)
```

I get only one error, but I expect the `example_2` field to give the same error.

Am I missing something, or is static type checking for default values of fields not supported?

Thanks!

**Comment by SupImDos** [comment, marked as answer]
I believe this is fixed in: #3430

See:
* https://github.com/pydantic/pydantic/issues/3429
* https://github.com/pydantic/pydantic/pull/3430

**Reply by samuelcolvin** [reply]
Thanks so much for updating this.
